src/components/App/App.jsx

```diff
@@ -12,10 +13,40 @@ import { allEncountersForUser } from '../../mockDataset.js';
 
 export default function App() {
   const [encounters, setEncounters] = useState([]);
+  const client = useApolloClient();
 
   useEffect(() => {
-    setEncounters(allEncountersForUser.data.queryUsers[0].encounters);
-  }, []);
+    const getEncounters = async () => {
+      try {
+        const { data } = await client.query({
+          query: gql`
+            query getEncounters($userName: String!) {
+              encounters(userName: $userName) {
+                  id
+                  userName
+                  encounterName
+                  partySize
+                  partyLevel
+                  summary
+                  description
+                  treasure
+                  encounterMonsters {
+                      monsterName
+                  }
+              }
+          }
+    `,
+          variables: {
+            "userName": "demo-many-encounters"
+          },
+        });
+        setEncounters(data.encounters);
+      } catch (error) {
+        console.error('Error fetching encounters: ', error);
+      }
+    };
+    getEncounters();
+  }, [client]);
```

Comment:
To make sure I am understanding the Apollo/GraphQL intersection correctly, this code block is being run every time the client changes, correct? as opposed to solely on the App component mount? Since the client is being defined as having the uri `https://tavern-keeper-be.onrender.com/graphql/`, how do we indicate that we want to hit a different endpoint, such as to query the monsters array, and does that update the value of the client (triggering this useEffect again)?

Reply:
You are correct that it runs on mount and if/when the client changes. The client is not the query, but the instance of `ApolloClient` that manages communication with the endpoint. In our API, the endpoint is always the same, so we don't need multiple clients. We don't change the endpoint, we change the query. So, instead of 
```
query: gql`
      query getEncounters($userName: String!) {
        encounters(userName: $userName) {
            id
            userName
            encounterName
            partySize
            partyLevel
            summary
            description
            treasure
            encounterMonsters {
                monsterName
            }
        }
      }
```

for querying all the monsters, we'll have
```
query: gql`
      query {
        monsters {
          monsterIndex
          monsterName
          size
          type
          alignment
          challengeRating
        }
      }
```

@tenthwalker Does that make sense?

Reply:
Ok, that makes sense now having seen the different calls side by side. Thanks for walking me through it!